crates/subspace-service/src/sync_from_dsn.rs

```diff
@@ -47,49 +53,72 @@ enum NotificationReason {
 /// Create node observer that will track node state and send notifications to worker to start sync
 /// from DSN.
 #[allow(clippy::too_many_arguments)]
-pub(super) fn create_observer_and_worker<Block, AS, Client, PG>(
+pub(crate) fn create_observer_and_worker<Block, AS, Client, PG, IQS, B>(
     segment_headers_store: SegmentHeadersStore<AS>,
     network_service: Arc<NetworkService<Block, <Block as BlockT>::Hash>>,
     node: Node,
     client: Arc<Client>,
-    mut import_queue_service: Box<dyn ImportQueueService<Block>>,
+    import_queue_service: Arc<Mutex<Box<IQS>>>,
     sync_target_block_number: Arc<AtomicU32>,
     pause_sync: Arc<AtomicBool>,
     piece_getter: PG,
+    fast_sync_enabled: bool,
+    fast_sync_state: Arc<parking_lot::Mutex<Option<NumberFor<Block>>>>,
+    sync_service: Arc<SyncingService<Block>>,
 ) -> (
     impl Future<Output = ()> + Send + 'static,
     impl Future<Output = Result<(), sc_service::Error>> + Send + 'static,
 )
 where
+    B: sc_client_api::Backend<Block>,
     Block: BlockT,
     AS: AuxStore + Send + Sync + 'static,
     Client: HeaderBackend<Block>
         + BlockBackend<Block>
         + BlockchainEvents<Block>
         + ProvideRuntimeApi<Block>
+        + ProofProvider<Block>
+        + LockImportRun<Block, B>
         + Send
         + Sync
+        + ClientExt<Block, B>
         + 'static,
-    Client::Api: SubspaceApi<Block, FarmerPublicKey>,
+    Client::Api: SubspaceApi<Block, FarmerPublicKey> + ObjectsApi<Block>,
     PG: DsnSyncPieceGetter + Send + Sync + 'static,
+    IQS: ImportQueueService<Block> + ?Sized + 'static,
 {
+    let network_service_clone = network_service.clone();
     let (tx, rx) = mpsc::channel(0);
+    let notification_sender = tx.clone();
     let observer_fut = {
         let node = node.clone();
         let client = Arc::clone(&client);
 
-        async move { create_observer(network_service.as_ref(), &node, client.as_ref(), tx).await }
+        async move {
+            create_observer(
+                network_service_clone.as_ref(),
+                &node,
+                client.as_ref(),
+                notification_sender,
+            )
+            .await
+        }
     };
     let worker_fut = async move {
         create_worker(
             segment_headers_store,
             &node,
-            client.as_ref(),
-            import_queue_service.as_mut(),
+            client.clone(),
+            import_queue_service,
```

Comment:
Why owned values? I think references should still work just fine.

Reply:
I can see that, but the question above remains

Reply:
There are type requirements down the stack.

Reply:
Which types? They were not required before, what has changed in a way that makes it mandatory?